src/Microsoft.Restier.Core/Conventions/ConventionBasedApiModelBuilder.cs

```diff
@@ -137,8 +136,7 @@ private IQueryable GetSingletonQuery(QueryExpressionContext context)
                 object target = null;
                 if (!singletonProperty.GetMethod.IsStatic)
                 {
-                    target = context.QueryContext.ApiContext
-                        .GetProperty(typeof(Api).AssemblyQualifiedName);
+                    target = context.QueryContext.GetApiService<IApi>();
```

Comment:
Personally I incline to keep a pure abstraction, but if you do think we should prevent another base implementation of Api then I agree with you.